src/LinearFractional.jl

```diff
@@ -1,130 +1,340 @@
+    T, field = ex.args[1], ex.args[2].value
+
+    T = esc(T)
+    fs = isexpr(fs, :tuple) ? map(esc, fs.args) : [esc(fs)]
+    :($([:($f(x::$T, args...) = (Base.@_inline_meta; $f(x.$field, args...)))
+        for f in fs]...);
+    nothing)
+end
+
+mutable struct LinearFractionalModel <: AbstractModel
+    model::JuMP.Model #JuMP.Model(solver=solver)
+    t::VariableRef
+    denominator_constraint::Union{Nothing, ConstraintRef}
+
+    function LinearFractionalModel(; kwargs...)
+        model = JuMP.Model(kwargs...)
+        model.ext[:is_linear_fractional] = true
+        t = @variable(model, lower_bound = 0.0, base_name = "lf_aux")
+        new(model, t, nothing)
     end
 end
 
+function LinearFractionalModel(optimizer_factory::JuMP.OptimizerFactory;
+               bridge_constraints::Bool=true, kwargs...)
+    model = LinearFractionalModel(; kwargs...)
+    JuMP.set_optimizer(model.model, optimizer_factory,
+                  bridge_constraints=bridge_constraints)
+    return model
+end
 
-storecontainerdata(m::LinearFractionalModel, variable, varname, idxsets, idxpairs, condition) =
-    m.transformedmodel.varData[variable] = JuMPContainerData(varname, map(collect,idxsets), idxpairs, condition)
+Base.broadcastable(model::LinearFractionalModel) = Ref(model)
 
+JuMP.object_dictionary(model::LinearFractionalModel) = model.model.obj_dict
 
-function AffExpr(vars::Vector{LinearFractionalVariable}, coeffs, constant)
-    t = vars[1].model.t
-    ys = [var.var for var in vars]
-    return LinearFractionalAffExpr(AffExpr(ys, coeffs, 0) + constant * t, t)
+# Wrap `VariableRef` instead of having fields, say
+# model::LinearFractionalModel and idx::Int
+# so that we can leverage existing `add_variable` and other methods
+# by simply passing the `vref`
+struct LinearFractionalVariableRef <: JuMP.AbstractVariableRef
+    model::LinearFractionalModel
+    vref::VariableRef
 end
 
+Base.copy(v::LinearFractionalVariableRef) = v
+Base.copy(v::LinearFractionalVariableRef, new_model::LinearFractionalModel) = LinearFractionalVariableRef(new_model, v.idx)
+
+Base.:(==)(v::LinearFractionalVariableRef, w::LinearFractionalVariableRef) = v.model === w.model && (v.vref == w.vref)
+Base.broadcastable(v::LinearFractionalVariableRef) = Ref(v)
+JuMP.isequal_canonical(v::LinearFractionalVariableRef, w::LinearFractionalVariableRef) = v == w
+JuMP.variable_type(::LinearFractionalModel) = LinearFractionalVariableRef
+function JuMP.add_variable(m::LinearFractionalModel, v::JuMP.AbstractVariable, name::String="")
 
-function addvariable(model::LinearFractionalModel, lb::Number, ub::Number, basename::String)
-    var = @variable(model.transformedmodel, basename=basename)
-    if !isinf(lb)# || !isinf(ub)
-        con = LinearConstraint(LinearFractionalAffExpr([var], [1], [0]), lb, ub)
-        addconstraint(model, con)
+    inner_vref = JuMP.add_variable(m.model, v, name)
+    vref = LinearFractionalVariableRef(m, inner_vref)
+
+    if has_lower_bound(inner_vref)
```

Comment:
I like how you have managed to leverage the function from JuMPExtension.jl imported line 20s and how you have used them to then be able to just have to create those tiny function made for vref like line 147